precompile/contracts/bibliophile/clearing_house.go

```diff
@@ -104,6 +113,99 @@ func getNotionalPositionAndMargin(stateDB contract.StateDB, input *GetNotionalPo
 	}
 }
 
+func getNotionalPositionAndRequiredMargin(stateDB contract.StateDB, input *GetNotionalPositionAndMarginInput) GetNotionalPositionAndMarginOutput {
+	positions, underlyingPrices, accountPreferences, activeMarketIds := getMarketsDataFromDB(stateDB, &input.Trader, input.Mode)
+	pendingFunding := big.NewInt(0)
+	if input.IncludeFundingPayments {
+		pendingFunding = getTotalFundingForCrossMarginPositions(stateDB, &input.Trader)
+	}
+	notionalPosition, margin, requiredMargin := hu.GetNotionalPositionAndRequiredMargin(
+		&hu.HubbleState{
+			Assets:         GetCollaterals(stateDB),
+			OraclePrices:   underlyingPrices,
+			ActiveMarkets:  activeMarketIds,
+		},
+		&hu.UserState{
+			Positions:      positions,
+			Margins:        getMargins(stateDB, input.Trader),
+			PendingFunding: pendingFunding,
+			AccountPreferences: accountPreferences,
+		},
+	)
+	return GetNotionalPositionAndMarginOutput{
+		NotionalPosition: notionalPosition,
+		Margin:           margin,
+		RequiredMargin:   requiredMargin,
+	}
+}
+
+func getCrossMarginAccountData(stateDB contract.StateDB, trader *common.Address, mode uint8, upgradeVersion hu.UpgradeVersion) GetTraderDataForMarketOutput {
+	positions, underlyingPrices, accountPreferences, activeMarketIds := getMarketsDataFromDB(stateDB, trader, mode)
+	notionalPosition, requiredMargin, unrealizedPnl := hu.GetCrossMarginAccountData(
+		&hu.HubbleState{
+			ActiveMarkets:  activeMarketIds,
+			OraclePrices:   underlyingPrices,
+			UpgradeVersion: upgradeVersion,
```

Comment:
dont need to accept upgrade version as a param. just send V2

Reply:
it not needed in the new code. removed it